src/mo_def/compUnit.ml

```diff
@@ -44,9 +44,9 @@ let comp_unit_of_prog as_lib (prog : prog) : comp_unit =
       assert (List.length tbs > 0);
       finish imports { it = ActorClassU (sp, tid, tbs, p, typ_ann, self_id, fields); note = d.note; at = d.at }
     (* let-bound terminal expressions *)
-    | [{it = LetD ({it = VarP i1; _}, ({it = ObjBlockE ({it = Type.Module; _}, fields); _} as e)); _}] when as_lib ->
+    | [{it = LetD ({it = VarP i1; _}, ({it = ObjBlockE ({it = Type.Module; _}, fields); _} as e), None); _}] when as_lib ->
       finish imports { it = ModuleU (Some i1, fields); note = e.note; at = e.at }
-    | [{it = LetD ({it = VarP i1; _}, e); _}] when is_actor_def e ->
+    | [{it = LetD ({it = VarP i1; _}, e, None); _}] when is_actor_def e ->
```

Comment:
Both of these worry me a little.
What if a library was written in desugared form with a fail clause
List.mo
```
let List = module { } else { loop {}}
```
If that is legal (and I think it is), will importing List.mo just crash.

Similarly,if the main actor file is authored as:


main.mo
```
let a = actor {} else {loop {}};
```
Will `moc main.mo` crash?

Reply:
blah